static/js/metrics-explorer.js

```diff
@@ -52,18 +52,36 @@ $(document).ready(function() {
 });
 
 
-function metricsExplorerDatePickerHandler(evt) {
+async function metricsExplorerDatePickerHandler(evt) {
     evt.preventDefault();
     $.each($(".range-item.active"), function () {
         $(this).removeClass('active');
     });
+    var selectedId = $(evt.currentTarget).attr("id");
     $(evt.currentTarget).addClass('active');
-    datePickerHandler($(this).attr('id'), "now", $(this).attr('id'))
+    datePickerHandler(selectedId, "now", selectedId);
+
+    const newMetricNames = await getMetricNames();
+    newMetricNames.metricNames.sort();
+  
+    $('.metrics').autocomplete('option', 'source', newMetricNames.metricNames);
+    
     // Update graph for each query
+   
     Object.keys(queries).forEach(async function(queryName) {
         var queryDetails = queries[queryName];
-        await getQueryDetails(queryName,queryDetails)
+        await getQueryDetails(queryName, queryDetails);
+        const tagsAndValue = await getTagKeyValue(queryDetails.metrics);
+        
+        queryDetails.everywhere = [];
+        queryDetails.everything = [];
+        availableEverywhere = tagsAndValue.availableEverywhere;
+        availableEverything = tagsAndValue.availableEverything[0];
+        const queryElement = $(`.metrics-query .query-name:contains(${queryName})`).closest('.metrics-query');
+        queryElement.find('.everywhere').autocomplete('option', 'source', availableEverywhere);
```

Comment:
Looks good! 
Just one thing: sort availableEverywhere and availableEverything before updating the options.